CMakeLists.txt

```diff
@@ -168,23 +170,35 @@ if(CPU_ONLY)
   target_compile_definitions(MaCh3CompileDefinitions INTERFACE CPU_ONLY)
 endif()
 
-if(MaCh3_GPU_ENABLED)
-  target_compile_definitions(MaCh3CompileDefinitions INTERFACE CUDA)
-  target_compile_definitions(MaCh3CompileDefinitions INTERFACE GPU_ON)
+if(MaCh3_GPU_ENABLED) 
+  target_compile_definitions(MaCh3GPUCompilerOptions INTERFACE CUDA)
+  target_compile_definitions(MaCh3GPUCompilerOptions INTERFACE GPU_ON)
 endif()
 
 if(NOT DEFINED MaCh3_LOW_MEMORY_STRUCTS_ENABLED)
   set(MaCh3_LOW_MEMORY_STRUCTS_ENABLED FALSE)
 endif()
 
 if(MaCh3_LOW_MEMORY_STRUCTS_ENABLED)
-  target_compile_definitions(MaCh3CompileDefinitions INTERFACE _LOW_MEMORY_STRUCTS_)
-endif()
+  target_compile_definitions(MaCh3CompilerOptions INTERFACE _LOW_MEMORY_STRUCTS_)
+  if(MaCh3_GPU_ENABLED)
+    # ETA - think the _LOW_MEMORY_STRUCTS_ also needs this definition
+    target_compile_definitions(MaCh3GPUCompilerOptions INTERFACE _LOW_MEMORY_STRUCTS_)
+  endif(MaCh3_GPU_ENABLED)
```

Comment:
Is it really neccessary? I think it should be fine wihtout but haven't tested

Reply:
removed this, you're right that it is not used.